### PR TITLE
Library api

### PR DESCRIPTION
**Pull request by juligasa**

### CODE REVIEW

**Review by juligasa — COMMENTED** [review]
Some preliminary comments

File: proto/entities/v1alpha/entities.proto [hunk 1]
```diff
@@ -297,3 +299,79 @@ message Mention {
...
+
+message ListLibraryRequest {
+  // Optional. Number of results per page. Default is defined by the server.
+  int32 page_size = 1;
+
+  // Optional. Value from next_page_token obtains from a previous response.
+  string page_token = 2;
+
+  // Optional. The query string to filter library results
+  string query = 3;
```
[the remark this answers is not in the document]
Comment (ericvicenti) [reply proto/entities/v1alpha/entities.proto:311]: for now just a string

File: proto/entities/v1alpha/entities.proto [hunk 2]
```diff
@@ -297,3 +299,79 @@ message Mention {
...
+  // alphabetical on the document title
+  ALPHABETICAL = 1;
+
+  // sort by the latest activity. (document change or comment)
+  LATEST_ACTIVITY = 2;
+}
+
+message EntityMetadata {
+  // Space/AccountUID of the document
+  string space = 1;
```
[the remark this answers is not in the document]
Comment (ericvicenti) [reply proto/entities/v1alpha/entities.proto:338]: whatever is fine!

[the remark this answers is not in the document]
Comment (ericvicenti) [reply proto/entities/v1alpha/entities.proto:338]: for now sure lets use account for consistency

File: proto/entities/v1alpha/entities.proto [hunk 3]
```diff
@@ -297,3 +299,79 @@ message Mention {
...
+  // when true, only return subscribed documents (including children documents of a recursive subscription)
+  bool filter_subscribed = 5;
+
+  // todo: add filter options based on "your" capabilities for the given document
+
+  // the frontend will filter these results further if the user has selected a 'favorite' filter, and will also do draft-based filtering
+}
+
+
+enum LibrarySort {
```
[the remark this answers is not in the document]
Comment (juligasa) [reply proto/entities/v1alpha/entities.proto:325]: The thing is that choosing a non stable sorting like any of the mentioned below, will cause pagination problems even if you don't change it between requests. The page token is tied to the index used to sort the values, if new records are inserted at not inserted at the end, we can skip records between calls. If we used a fixed index internally and then reorder according to the sorting algorithm, then you could get disordered records between calls (if new events are getting in)

File: proto/entities/v1alpha/entities.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (ericvicenti) [reply proto/entities/v1alpha/entities.proto:311]: same as search API! if I send a query string like "fo", then a document called "Foundation Series" would appear, or any document with a title that includes that string (case insensitive)

File: proto/entities/v1alpha/entities.proto (hunk 3 above)
[the remark this answers is not in the document]
Comment (ericvicenti) [reply proto/entities/v1alpha/entities.proto:325]: oh, well can you return a pagination token that somehow provides stability? from the front end it is an opaque string, so you can encode other information in there such as sort order

[the remark this answers is not in the document]
Comment (juligasa) [reply proto/entities/v1alpha/entities.proto:325]: Should I throw an error then if the sorting algo embedded in the token does not match the sorting algo of the request? @ericvicenti ?

[the remark this answers is not in the document]
Comment (ericvicenti) [reply proto/entities/v1alpha/entities.proto:330]: Yes that would be fine I think!

File: proto/entities/v1alpha/entities.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (ericvicenti) [reply proto/entities/v1alpha/entities.proto:316]: yep its a list

**Pull request closed without merging**